code/game/objects/items/surgery_tray.dm

```diff
@@ -0,0 +1,338 @@
+/**
+ * surgery trays
+ *
+ * a storage object that displays tools in its contents, and can be folded up and carried. click it to draw a random tool
+ *
+ */
+
+
+/datum/storage/medicart
+	max_total_storage = 30
+	max_specific_storage = WEIGHT_CLASS_NORMAL
+	max_slots = 14
+
+/datum/storage/medicart/New()
+	. = ..()
+	set_holdable(list(
+		/obj/item/blood_filter,
+		/obj/item/bonesetter,
+		/obj/item/cautery,
+		/obj/item/circular_saw,
+		/obj/item/clothing/mask/surgical,
+		/obj/item/hemostat,
+		/obj/item/razor,
+		/obj/item/retractor,
+		/obj/item/scalpel,
+		/obj/item/stack/medical/bone_gel,
+		/obj/item/stack/sticky_tape/surgical,
+		/obj/item/surgical_drapes,
+		/obj/item/surgicaldrill,
+	))
+
+/obj/item/surgery_tray
+	name = "surgery tray"
+	desc = "A Deforest brand medical cart. It is a folding model, meaning the wheels on the bottom can be retracted and the body used as a tray."
+	icon = 'icons/obj/medicart.dmi'
+	icon_state = "tray"
+	w_class = WEIGHT_CLASS_BULKY
+	slowdown = 1
+	item_flags = SLOWS_WHILE_IN_HAND
+
+	var/tray_toggled = TRUE
+
+	var/obj/item/stack/medical/bone_gel/bone_gel
+	var/obj/item/stack/sticky_tape/surgical/surgical_tape
+	var/obj/item/blood_filter/blood_filter
+
+	var/obj/item/razor/surgery/razor
+	var/obj/item/bonesetter/bonesetter
+	var/obj/item/surgical_drapes/surgical_drapes
+
+	var/obj/item/surgicaldrill/surgical_drill
+	var/obj/item/cautery/cautery
+	var/obj/item/circular_saw/circular_saw
+	var/obj/item/hemostat/hemostat
+	var/obj/item/retractor/retractor
+	var/obj/item/scalpel/scalpel
+
+/obj/item/surgery_tray/deployed
+	tray_toggled = FALSE
+
+/obj/item/surgery_tray/Initialize(mapload)
+	. = ..()
+	create_storage(storage_type = /datum/storage/medicart)
+	PopulateContents()
+	AddElement(/datum/element/noisy_movement)
+	AddElement(/datum/element/drag_pickup)
+	set_tray_mode(tray_toggled)
+	register_context()
+
+/obj/item/surgery_tray/Destroy(force)
```

Comment:
For bonus points, you may want to implement `/obj/item/surgery_tray/deconstruct()` so destroying it ICly (melting via acid, explosion, fire? idk) drops any tools to the ground rather than voiding them

Reply:
this one was fun. tested with c4
![image](https://github.com/tgstation/tgstation/assets/110322848/9d049246-85c9-495a-811d-b5708645d6b8)
and a flamethrower
![image](https://github.com/tgstation/tgstation/assets/110322848/8b9cdd72-05c0-4c95-a6f8-b49c5a1d5d25)